saleor/product/search.py

```diff
@@ -1,33 +1,29 @@
 from collections import defaultdict
+from collections.abc import Iterable
 from typing import TYPE_CHECKING, Union
 
 from django.conf import settings
 from django.contrib.postgres.search import SearchQuery, SearchRank
 from django.db.models import F, Q, Value, prefetch_related_objects
-from django.db.models.expressions import Exists, OuterRef
 
 from ..attribute import AttributeInputType
-from ..attribute.models import (
-    AssignedProductAttributeValue,
-    Attribute,
-    AttributeProduct,
-)
+from ..attribute.models import Attribute
 from ..core.postgres import FlatConcatSearchVector, NoValidationSearchVector
 from ..core.utils.editorjs import clean_editor_js
-from .models import Product
+from ..product.models import Product
 
 if TYPE_CHECKING:
     from django.db.models import QuerySet
 
 PRODUCT_SEARCH_FIELDS = ["name", "description_plaintext"]
 PRODUCT_FIELDS_TO_PREFETCH = [
     "variants__attributes__values",
-    "variants__attributes",
+    "variants__attributes__assignment__attribute",
     "attributevalues__value",
     "product_type__attributeproduct__attribute",
 ]
 
-PRODUCTS_BATCH_SIZE = 300
+PRODUCTS_BATCH_SIZE = 100
 # Setting threshold to 300 results in about 350MB of memory usage
```

Comment:
The comment should be updated as well, when the batch size is chaning

Reply:
Right. Fixed it. Currently, during my tests there it is 767MB of memory usage. On main the same data set it was over 1600MB.